File: src/main/java/org/spilth/search/SearchCommand.java
package org.spilth.search;

import com.beust.jcommander.Parameter;
import com.beust.jcommander.Parameters;
import org.spilth.validators.FormatValidator;

import java.util.List;

@Parameters(commandNames = "search", commandDescription = "Searches Maven Central for dependencies")
public class SearchCommand {
    @Parameter(description = "Search term", arity = 1)
    private List<String> searchTerms;

    @Parameter(names = {"--format", "-f"}, description = "Dependency output format", validateWith = FormatValidator.class)
    private String format = "maven";

    @Parameter(names = {"--help", "-h"}, help = true)
    private boolean help = false;

    public String getFormat() {
        return format;
    }

    public String getSearchTerm() {
        return searchTerms.get(0);
    }

    public boolean isHelp() {
        return help;
    }
}

File: src/main/kotlin/org/spilth/search/SearchCommand.kt
package org.spilth.search

import com.beust.jcommander.Parameter
import com.beust.jcommander.Parameters
import org.spilth.validators.FormatValidator

@Parameters(commandNames = ["search"], commandDescription = "Searches Maven Central for dependencies")
class SearchCommand {
    @Parameter(description = "Search Term", required = true)
    var searchTerm: String? = null

    @Parameter(names = ["--format", "-f"], description = "Dependency output format: maven (default) or gradle", validateWith = [FormatValidator::class])
    var format = "maven"

    @Parameter(names = ["--help", "-h"], description = "Show help", help = true)
    var isHelp = false
}
